Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -23,57 +28,86 @@ final class SchemesGenerator: SchemesGenerating {
-        let schemesDirectory = try createSchemesDirectory(projectPath: generatedProject.path, shared: scheme.shared)
-        let schemePath = schemesDirectory.appending(component: "\(scheme.name).xcscheme")
-
-        let generatedBuildAction = schemeBuildAction(scheme: scheme, project: project, generatedProject: generatedProject)
-        let generatedTestAction = schemeTestAction(scheme: scheme, project: project, generatedProject: generatedProject)
-        let generatedLaunchAction = schemeLaunchAction(scheme: scheme, project: project, generatedProject: generatedProject)
-        let generatedProfileAction = schemeProfileAction(scheme: scheme, project: project, generatedProject: generatedProject)
-        let generatedArchiveAction = schemeArchiveAction(scheme: scheme, project: project, generatedProject: generatedProject)
+    ///     - scheme: Project scheme.
+    ///     - xcPath: Path to workspace's .xcworkspace or project's .xcodeproj.
+    ///     - path: Path to workspace or project folder.
+    ///     - graph: Tuist graph.
+    ///     - generatedProjects: Project paths mapped to generated projects.
+    private func generateScheme(scheme: Scheme,
+                                xcPath: AbsolutePath,
+                                path: AbsolutePath,
+                                graph: Graphing,
+                                generatedProjects: [AbsolutePath: GeneratedProject]) throws {
+        let schemeDirectory = try createSchemesDirectory(path: xcPath, shared: scheme.shared)
+        let schemePath = schemeDirectory.appending(component: "\(scheme.name).xcscheme")
+        let generatedBuildAction = try schemeBuildAction(scheme: scheme,
+                                                         graph: graph,
+                                                         rootPath: path,
+                                                         generatedProjects: generatedProjects)
+        let generatedTestAction = try schemeTestAction(scheme: scheme,
+                                                       graph: graph,
+                                                       rootPath: path,
+                                                       generatedProjects: generatedProjects)
+        let generatedLaunchAction = try schemeLaunchAction(scheme: scheme,
+                                                           graph: graph,
+                                                           rootPath: path,
+                                                           generatedProjects: generatedProjects)
+        let generatedProfileAction = try schemeProfileAction(scheme: scheme,
+                                                             graph: graph,
+                                                             rootPath: path,
+                                                             generatedProjects: generatedProjects)
+        
+        let generatedArchiveAction = try schemeArchiveAction(scheme: scheme,
```

Comment:
super minor: Seeing the order of definition in `XCScheme` is analyze then archive may be worth being consistent here too?